### ISSUE TITLE
Irrelevant UnusedPrivateMember warnings on detekt 2.0.0 from IJ plugin

### ISSUE THREAD

**Issue by kevinguitar** [detekt/detekt#8949]
Hello 👋 

After updating to detekt 2.0.0-alpha.1, we noticed an irrelevant warning in the IDE (Android Studio)

```kotlin
@Composable
fun Foo() = Unit

@Preview
@Composable
private fun Foo_Preview() = Foo()
```

This is the rule setting
<img width="305" height="111" alt="Image" src="https://github.com/user-attachments/assets/743dfdcf-8c1f-445d-94a1-d697b4c92c82" />

Confirmed that in Settings -> Tools -> Detekt, the config location is set correctly.

## Expected Behavior
Detekt should not report this if the environment is using detekt 2.x?

## Observed Behavior
Detekt reports `UnusedPrivateMember` on `Foo_Preview`
<img width="1460" height="582" alt="Image" src="https://github.com/user-attachments/assets/819f7ce7-6ad3-46df-aebb-88f548cc0877" />

## Your Environment
* Version of detekt used: 2.0.0-alpha.1
* Version of detekt IJ plugin used: 2.4.3
* Version of Gradle used (if applicable): Gradle 9.2 (AGP 8.13.2)
